traffic_ops/app/db/migrations/2021081800000001_no_null_username.up.sql

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+ALTER TABLE ONLY public.tm_user
+ALTER COLUMN username
+SET NOT NULL;
```

Comment:
Should we set the `null` usernames to a blank string like we are for `description`, before we add the constraint?

Reply:
no, I want it to fail intentionally, since a NULL username is something very broken that needs attention - a blank string username is almost certainly not what you actually want.

Besides that, usernames have to be unique - unless they're null. So if more than one user has a NULL username that coalesce won't work anyway.